pkg/controllers/jobset_controller.go

```diff
@@ -86,7 +102,22 @@ func (r *JobSetReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctr
 		return ctrl.Result{}, client.IgnoreNotFound(err)
 	}
 
-	log := ctrl.LoggerFrom(ctx).WithValues("jobset", klog.KObj(&js))
+	// Track JobSet status updates that should be performed at the end of the reconciliation attempt.
+	updateStatusOpts := statusUpdateOpts{}
+
+	// Reconcile the JobSet.
+	result, err := r.reconcile(ctx, &js, &updateStatusOpts)
+	if err != nil {
+		return result, err
+	}
+
+	// At the end of this Reconcile attempt, do one API call to persist all the JobSet status changes.
+	return ctrl.Result{}, r.updateJobSetStatus(ctx, &js, &updateStatusOpts)
```

Comment:
```suggestion
	return ctrl.Result{RequeueAfter: result.RequeueAfter}, r.updateJobSetStatus(ctx, &js, &updateStatusOpts)
```

This should fix the failing test, as it currently isn't setting the `RequeueAfter` and the JobSet doesn't get requeued after TTL expiration and we the TTL logic is never executed.

Reply:
Thanks @dejanzele! I will test this out locally and update the PR If so

Reply:
It worked, thanks!